tests/unit/test_lazy_load_agate.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+import sys
+
+
+@pytest.fixture
+def remove_agate_from_path():
+    original_sys_modules = sys.modules.copy()
+
+    # conftest loads agate modules so we need to remove them
+    # and this file
+    modules_to_remove = [m for m in sys.modules if "agate" in m]
```

Comment:
Doing this first in the fixture should make the test always pass since the logic for the check is the same as the logic for setup. Have you considered omitting this fixture to reset the modules, and instead verifying this logic is correct by checking for a module that should be installed (e.g. `dbt-adapters`)?